layerindex/tools/fedora-fetch.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+
+# Fedora Pagure (e.g. https://src.fedoraproject.org/) fetch utility
+
+# Copyright (C) 2017, 2018 Intel Corporation
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+import sys
+import os
+import argparse
+import json
+import requests
+import re
+import subprocess
+
+
+def get_repos_api(url, toplevel_item=None, name_attr='name', use_link_header=True, link_item='pagination', link_next_attr='next'):
+    link_re = re.compile('<(http[^>]+)>; rel="([a-zA-Z0-9]+)"')
+    repos = None
+    while True:
+        session = requests.Session()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *session*. The resource is allocated by call *sessions.Session*. Currently, there are execution paths that do not contain closure statements, for example, when an else condition at line 51 evaluates to true. To prevent this resource leak, close *session* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).